src/main/kotlin/domain/Board.kt

```diff
@@ -18,6 +19,40 @@ class Board(
 
     fun getNearByMineCount(height: Int, width: Int): Int {
         val coordinate = Coordinate(height, width)
-        return fields.getNearByFields(coordinate).filterIsInstance<Mine>().count()
+        return fields.getNearByFields(coordinate).count { it.field is Mine }
+    }
+
+    fun isMine(height: Int, width: Int): Boolean {
+        val coordinate = Coordinate(height, width)
+        return fields.getField(coordinate) is Mine
+    }
+
+    fun isGameOver(): Boolean {
+        return fields.isLandAllOpened()
+    }
+
+    tailrec fun open(height: Int, width: Int) {
+        // 해당 좌표 열기
+        val coordinate = Coordinate(height, width)
+        fields.open(coordinate)
+
+        // 인근에 지뢰가 아닌 좌표 열기
+        getNotOpenedLandAndHasNoMineWithCoordinate(coordinate).forEach { land ->
+            open(land.coordinate.row, land.coordinate.col)
+        }
+    }
+
+    private fun getNotOpenedLandAndHasNoMineWithCoordinate(coordinate: Coordinate): List<FieldWithCoordinate> {
+        if (hasMineNearby(coordinate))
+            return emptyList()
+        return getNotOpenedLand(coordinate)
+    }
+
+    private fun hasMineNearby(coordinate: Coordinate): Boolean {
+        return fields.getNearByFields(coordinate).count { it.field is Mine } > 0
+    }
+
+    private fun getNotOpenedLand(coordinate: Coordinate): List<FieldWithCoordinate> {
+        return fields.getNearByFields(coordinate).filter { it.field is Land && it.field.isOpened.not() }
```

Comment:
```suggestion
        return fields.getNearByFields(coordinate).filter { it.field is Land && !it.field.isOpened }
```
not은 연산자 함수이기 때문에 이렇게도 표현할 수 있습니다. 혹시 구현해주신 방식을 의도해주신 걸까요?

Reply:
!는 생각보다 보기 놓치기 쉽다고 생각해서 not을 통해 조금 더 명시적으로 표현하기 위해 의도한 부분이에요~!